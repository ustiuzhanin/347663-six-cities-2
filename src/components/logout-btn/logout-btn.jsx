import React from "react";
import PropTypes from "prop-types";

import { connect } from "react-redux";
import { ActionCreator as AuthActionCreator } from "../../reducer/auth/auth";
import { ActionCreator } from "../../reducer/user/user";

const LogoutBtn = (props) => {
  const logoutClickHandler = () => {
    const { logoutUser } = props;

    localStorage.removeItem("token");
    logoutUser();
  };

  return (
    <li className="header__nav-item user">
      <button
        className="header__nav-link header__nav-link--btn"
        type="button"
        onClick={logoutClickHandler}
      >
        <div className="header__nav-link--logout"></div>
        <div className="visually-hidden">Logout</div>
      </button>
    </li>
  );
};

LogoutBtn.propTypes = {
  logoutUser: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  logoutUser: () => {
    dispatch(AuthActionCreator.logoutUser());
    dispatch(ActionCreator.clearUser());
    dispatch(AuthActionCreator.requireAuthorization(true));
  },
});

export default connect(null, mapDispatchToProps)(LogoutBtn);
