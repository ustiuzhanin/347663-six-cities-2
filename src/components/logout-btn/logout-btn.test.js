import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import { LogoutBtn } from "./logout-btn.jsx";

test(`LogoutBtn's snapshot`, () => {
  const renderer = new ShallowRenderer();

  renderer.render(<LogoutBtn logoutUser={jest.fn()} />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
