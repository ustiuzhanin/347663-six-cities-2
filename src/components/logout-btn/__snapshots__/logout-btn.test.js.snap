// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutBtn's snapshot 1`] = `
<li
  className="header__nav-item user"
>
  <button
    className="header__nav-link header__nav-link--btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="header__nav-link--logout"
    />
    <div
      className="visually-hidden"
    >
      Logout
    </div>
  </button>
</li>
`;
